### ISSUE TITLE
Compiler generates 8-bit instead of 16-bit LD instruction

### ISSUE THREAD

**Issue by ghost** [HSAFoundation/HLC-HSAIL-Development-LLVM#22]
I'm using the latest `llc` from the branch `hsail-stable-3.7` and the latest `CLOC.sh` from the master branch of the `CLOC` repository.

Consider the following OpenCL program:

```
__kernel void Bug2(__global unsigned short *a, __global long *b)
{
    *b = *a;
}
```

Compiling this program with the `CLOC.sh` gives the following HSAIL code:

```
module &__llvm_hsail_module:1:0:$full:$large:$near;

prog kernel &__OpenCL_Bug2_kernel(
    kernarg_u64 %__global_offset_0,
    kernarg_u64 %__global_offset_1,
    kernarg_u64 %__global_offset_2,
    kernarg_u64 %__printf_buffer,
    kernarg_u64 %__vqueue_pointer,
    kernarg_u64 %__aqlwrap_pointer,
    kernarg_u64 %a,
    kernarg_u64 %b)
{
    // BB#0:
    ld_kernarg_align(8)_width(all)_u64  $d0, [%a];
    ld_global_align(2)_u8   $s0, [$d0];
    cvt_u64_u32 $d0, $s0;
    ld_kernarg_align(8)_width(all)_u64  $d1, [%b];
    st_global_align(8)_u64  $d0, [$d1];
    ret;
};
```

The second load instruction in the kernel (`ld_global_align(2)_u8`) loads only a single byte instead of the whole `unsigned short` from memory. Interestingly enough, when changing the type of `b` to e.g. `int`, the correct 16-bit load instruction is generated.

**Comment by mzimmer** [comment]
Is a fix to this problem as simple as changing `MVT::i8` to `LD->getMemoryVT()` on line 1785:
https://github.com/HSAFoundation/HLC-HSAIL-Development-LLVM/blob/hsail-stable-3.7/lib/Target/HSAIL/HSAILISelLowering.cpp#L1785?

**Comment by arsenm** [comment]
Fixed in hsail-review-v4 and hsail-stable-3.7
